### PR TITLE
Split prover and verifier in different files + simpler fetch method

### PR DESCRIPTION
**Pull request by heeckhau**
Split prover and verifier in different files + simpler fetch method

### CODE REVIEW

File: wasm/prover/tests/web.rs [hunk 1]
```diff
@@ -18,6 +20,19 @@ macro_rules! log {
...
 
+#[wasm_bindgen_test]
+async fn test_fetch() {
+    let url = "https://swapi.info/api/";
+    let mut opts = RequestInit::new();
+    opts.method("GET");
+
+    let rust_string: String = tlsn_extension_rs::fetch_as_json_string(&url, &opts)
+        .await
+        .expect("TODO");
```
Comment (th4s) [inline wasm/prover/tests/web.rs:31]: In tests it is ok to use `unwrap()`.

File: wasm/prover/src/lib.rs [hunk 2]
```diff
@@ -28,18 +28,12 @@ pub(crate) use log;
...
-    let request = WebsysRequest::new_with_str_and_init(url, opts)?;
-    let promise = fetch(&request);
-    let future = JsFuture::from(promise);
-    let resp_value = future.await?;
-    let resp: Response = resp_value.dyn_into()?;
+    let request = Request::new_with_str_and_init(url, opts)?;
+    let window = web_sys::window().unwrap();
+    let resp_value = JsFuture::from(window.fetch_with_request(&request)).await?;
+    assert!(resp_value.is_instance_of::<Response>());
+    let resp: Response = resp_value.dyn_into().unwrap();
```
Comment (th4s) [inline wasm/prover/src/lib.rs:36]: Was there a reason to use `unwrap()` here instead of `?`

File: wasm/prover/src/prover.rs [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+    // let perf_layer = performance_layer()
+    //     .with_details_from_fields(Pretty::default());
+
+    // tracing_subscriber::registry()
+    //     .with(tracing_subscriber::filter::LevelFilter::DEBUG)
+    //     .with(fmt_layer)
+    //     .with(perf_layer)
+    //     .init(); // Install these as subscribers to tracing events
+
+    // https://github.com/rustwasm/console_error_panic_hook
```
Comment (th4s) [inline wasm/prover/src/prover.rs:108]: Is it important to keep this? Usually, I prefer to remove out-commented code as it is in git anyways.

File: wasm/prover/src/prover.rs [hunk 4]
```diff
@@ -0,0 +1,482 @@
...
+fn log_phase(phase: ProverPhases) {
+    log!(
+        "!@# tlsn-js {}: {}",
+        phase as u8,
+        phase.get_message().unwrap()
+    );
+}
+
+#[wasm_bindgen]
+pub async fn prover(
```
Comment (th4s) [inline wasm/prover/src/prover.rs:75]: At some point we should invest the effort and break this down into smaller functions.

**Review by th4s — COMMENTED** [review]
Looking good, some comments :+1:

File: wasm/prover/src/prover.rs (hunk 3 above)
Comment (heeckhau) [reply wasm/prover/src/prover.rs:108]: I agree. But we'll probably enable this again soon.

File: wasm/prover/src/prover.rs (hunk 4 above)
Comment (heeckhau) [reply wasm/prover/src/prover.rs:75]: Indeed. But waiting on @0xtsukino's Extism experiments first

**Pull request merged**